flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergSink.java

```diff
@@ -530,7 +586,26 @@ IcebergSink build() {
           flinkWriteConf,
           equalityFieldIds,
           flinkWriteConf.branch(),
-          overwriteMode);
+          overwriteMode,
+          createCompactionProperties(writeOptions, readableConfig.toMap()));
+    }
+
+    /**
+     * Create the properties for the compaction. The configuration from SQL hints has a higher
+     * priority than that from SET statements.
+     *
+     * @param writeProperties config from SQL Hints. Height level.
+     * @param setProperties config from SET statement. Low level.
+     * @return config of compaction
+     */
+    private Map<String, String> createCompactionProperties(
```

Comment:
As mentioned above, I would prefer `FlinkMaintenanceConfig` object instead of map.